tests/foreman/api/test_repository.py

```diff
@@ -1153,6 +1157,47 @@ def test_positive_list_puppet_modules_with_multiple_repos(self):
         # Verify that number of modules from the first repo has not changed
         self.assertEqual(modules_num, len(repo1.puppet_modules()['results']))
 
+    @tier2
+    @upgrade
+    def test_positive_access_protected_repository(self):
+        """Access protected/https repository data file URL using organization
+        debug certificate
+
+        :id: 4dba5b31-1818-45dd-a9bd-3ec627c3db57
+
+        :expectedresults: The repository data file successfully accessed.
+
+        :BZ: 1242310
+
+        :CaseImportance: Integration
+        """
+        # create a new protected repository
+        repository_data = entities.Repository(
+            url=FAKE_2_YUM_REPO,
+            content_type=REPO_TYPE['yum'],
+            product=self.product,
+            unprotected=False,
+        ).create().read_json()
+        repo_data_file_url = urljoin(
+            repository_data['full_path'], 'repodata/repomd.xml')
```

Comment:
Is there a field `full_path`? Why not adding that field to nailgun & use `read()` instead of `read_json()` then?

Reply:
This is a read only field, there are many read only fields there, think using read_json is OK as it's a public method, but we should update nailgun to save the read_json data (to not make additional requests) and to check the field  in the saved json data  if the attribute does not exist.